internal/gps/vcs_repo.go

```diff
@@ -31,6 +31,25 @@ type gitRepo struct {
 	*vcs.GitRepo
 }
 
+func newGitRepo(ustr, path string) (*gitRepo, error) {
+	r, err := vcs.NewGitRepo(ustr, path)
+	if err != nil {
+		if _, ok := err.(*vcs.LocalError); ok {
+			// if vcs could not initialize the repo due to a local error
+			// then the local repo is in an incorrect state. Remove and
+			// treat it as a new not-yet-cloned repo.
+			os.RemoveAll(path)
```

Comment:
We should at least log something IMHO. If somebody runs into this case they're already in a very strange state, and running dep with `-v` should output as much as possible for why things aren't going "right". Not sure how easy it would be to plumb the logger in, so maybe not worth it in this PR, just generally get scared of widely-used tools that silently drop errors.